packages/react/src/use-action-form/utils.ts

```diff
@@ -365,7 +388,7 @@ export const reshapeDataForGraphqlApi = async (client: AnyClient, defaultValues:
 
       const inputHasId = "id" in input;
       const inputHasMoreFields = Object.keys(input).filter((field) => field !== "__typename").length > 1;
-      const inputUpdateId = updates[path!]?.[0];
+      const inputUpdateId = idPath && recordIdsAtPath[idPath] ? recordIdsAtPath[idPath][0] : undefined;
```

Comment:
Consider adding a check for `recordIdsAtPath[idPath].length` in the off chance that `recordIdsAtPath[idPath]` is an empty array

Reply:
:+1: